libqrexec/exec.c

```diff
@@ -264,7 +282,7 @@ static int load_service_config_raw(struct qrexec_parsed_command *cmd,
 
     int ret = find_file(config_path, cmd->service_descriptor,
                         config_full_path, sizeof(config_full_path), NULL);
-    if (ret < 0 && strcmp(cmd->service_descriptor, cmd->service_name) != 0)
+    if (ret == -1 && cmd->service_descriptor != cmd->service_name)
```

Comment:
Why? I don't think your new code makes them point at the same buffer ever. But also, this looks like assumption the strings share the buffer if they are the same, which isn't even true.

Reply:
This is a relic of an earlier draft that _did_ make this guarantee.  Now fixed.